framework/src/executioners/Steady.C

```diff
@@ -22,108 +22,20 @@ template <>
 InputParameters
 validParams<Steady>()
 {
-  return validParams<Executioner>();
-}
+  auto params = validParams<Transient>();
 
-Steady::Steady(const InputParameters & parameters)
-  : Executioner(parameters),
-    _problem(_fe_problem),
-    _time_step(_problem.timeStep()),
-    _time(_problem.time()),
-    _final_timer(registerTimedSection("final", 1))
-{
-  _problem.getNonlinearSystemBase().setDecomposition(_splitting);
+  params.set<unsigned int>("num_steps") = 1;
```

Comment:
This is not enough. We will need to hide lots of parameters that does not make sense for steady-state solves. Another way could be moving stuff currently in `Transient` into `Executioner` and let both `Transient` and `Steady` derive from it.

Reply:
I like that idea and thought of it yesterday - for now I was just trying out something simple to see if it would even work.

The MOOSE team still hasn't even really decided that we want to do what's in this PR - but if we do want to go forward with it then I think that makes sense.